importers/src/bible/osis/index.ts

```diff
@@ -0,0 +1,945 @@
+                    title: tag.attributes.osisIDWork,
+                    chapterVerseSeparator: ':',
+                    ...this.options.versionMeta
+                };
+                break;
+            }
+            case OsisXmlNodeName.WORK: {
+                // work is handled in text parsing
+                break;
+            }
+            case OsisXmlNodeType.BOOK:
+                const bookMeta =
+                    this.options.bookMeta &&
+                    tag.attributes.osisID &&
+                    this.options.bookMeta.get(tag.attributes.osisID);
+                // currently we force bookMeta to be provided by the user (since the osis
+                // file we are currently working on doesn't have the data in the file)
+                if (!bookMeta)
+                    throw new Error(`book metadata missing for ${tag.attributes.osisID}`);
+
+                const bookNr = context.books.length + 1;
+
+                context.currentBook = {
+                    osisId: tag.attributes.osisID!,
+                    type: bookNr < 40 ? 'ot' : 'nt',
+                    number: bookNr,
+                    ...bookMeta
+                };
+
+                context.contentContainerStack = [{ type: 'root', contents: [] }];
+                context.sectionStack = [];
+                context.currentChapter = 0;
+                context.currentVerse = 0;
+                break;
+            case OsisXmlNodeName.CHAPTER: {
+                if (typeof context.currentChapter !== 'number')
+                    throw new Error(`chapter outside a book`);
+                if (!tag.attributes.osisID)
+                    throw new Error(
+                        this.getErrorMessageWithContext('chapter tag without osisID', context)
+                    );
+                const numbers = tag.attributes.osisID.split('.');
+                context.currentChapter++;
+                if (+numbers[1] !== context.currentChapter)
+                    throw new Error(`chapter number mismatch ${tag.attributes.osisID}`);
+                // sometimes there is a verse 1 tag, sometimes not
+                context.currentVerse = 1;
+                break;
+            }
+            case OsisXmlNodeName.VERSE: {
+                if (typeof context.currentVerse !== 'number') {
+                    throw new Error(`verse outside a chapter`);
+                }
+                if (!tag.attributes.osisID)
+                    throw new Error(
+                        this.getErrorMessageWithContext('verse tag without osisID', context)
+                    );
+
+                const numbers = tag.attributes.osisID.split('.');
+                // sometimes there is a verse 1 tag, sometimes not. sometimes the same verse number
+                // has two verse tags when a section splits a verse
+                if (+numbers[2] !== 1 && +numbers[2] !== context.currentVerse)
+                    context.currentVerse++;
```

Comment:
https://eslint.org/docs/rules/no-plusplus

Reply:
i think i don't agree with eslint there. i like ++ for what it is. you?